### ISSUE TITLE
Dask_Jobqueue not getting resources allocated when works are spawned

### ISSUE THREAD

**Issue by kyrubas** [dask/dask-jobqueue#289]
I'm trying to use dask_jobqueue on our SLUMRCluster, but having some problems. I can create the SLURMCluster object and spawn workers as show via the squeue command, but when inspecting the SLURMCluster object a second time, no cores or memory is allocated to it and I am unable to get any submitted jobs to return.

Here's an example of what's going on:

```
In [1]: from dask_jobqueue import SLURMCluster

In [2]: cluster = SLURMCluster(cores=1, memory='1GB', project='testing', queue='thequeue')

In [3]: !squeue
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
            224881    thequeue     bash usera  R    1:43:21      1 svrhpc3

In [4]: print(cluster)
SLURMCluster(cores=0, memory=0 B, workers=0/0, jobs=0/0)

In [5]: cluster.scale(4)

In [6]: !squeue
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
            225031    thequeue dask-wor userb  R       0:05      1 svrhpci
            225032    thequeue dask-wor userb  R       0:05      1 svrhpci
            225033    thequeue dask-wor userb  R       0:05      1 svrhpci
            225034    thequeue dask-wor userb  R       0:05      1 svrhpci
            224881    thequeue     bash usera  R    1:43:38      1 svrhpc3

In [7]: print(cluster)
SLURMCluster(cores=0, memory=0 B, workers=0/4, jobs=0/4)

```

**Comment by guillaumeeb** [comment]
Hi @kyrubas, thanks for using Dask and Dask-jobqueue!

See https://jobqueue.dask.org/en/latest/debug.html#interact-with-your-job-queuing-system, are you able to get any log or output from the workers jobs? Do you see some problems like worker not being able to connect to scheduler? There are often some networks problem involved with this kind of issues.

**Comment by kyrubas** [comment]
Thanks for your time @guillaumeeb 
So here's the output that I'm getting from the workers. It appears to be hanging then timing: 
`distributed.nanny - INFO -         Start Nanny at: 'tcp://X.X.X.X:AAAAA'
distributed.worker - INFO -       Start worker at:    tcp://X.X.X.X:BBBBB
distributed.worker - INFO -          Listening to:    tcp://X.X.X.X:BBBBB
distributed.worker - INFO -          dashboard at:          X.X.X.X:CCCCC
distributed.worker - INFO - Waiting to connect to:   tcp://Y.Y.Y.Y:DDDDD
distributed.worker - INFO - -------------------------------------------------
distributed.worker - INFO -               Threads:                          1
distributed.worker - INFO -                Memory:                 1000.00 MB
distributed.worker - INFO -       Local Directory: /homes/user/data/dasktest/worker-mnwqz6u0
distributed.worker - INFO - -------------------------------------------------
distributed.worker - INFO - Waiting to connect to:   tcp://Y.Y.Y.Y:DDDDD
distributed.worker - INFO - Waiting to connect to:   tcp://Y.Y.Y.Y:DDDDD
distributed.worker - INFO - Waiting to connect to:   tcp://Y.Y.Y.Y:DDDDD
distributed.worker - INFO - Waiting to connect to:   tcp://Y.Y.Y.Y:DDDDD
distributed.worker - INFO - Waiting to connect to:   tcp://Y.Y.Y.Y:DDDDD
distributed.nanny - INFO - Closing Nanny at 'tcp://X.X.X.X:AAAAA'
distributed.worker - INFO - Stopping worker at tcp://X.X.X.X:BBBBB
distributed.nanny - ERROR - Timed out connecting Nanny '<Nanny: None, threads: 1>' to scheduler 'tcp://Y.Y.Y.Y:DDDDD'
Traceback (most recent call last):
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/distributed/nanny.py", line 303, in instantiate
    timedelta(seconds=self.death_timeout), self.process.start()
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/tornado/gen.py", line 1133, in run
    value = future.result()
tornado.util.TimeoutError: Timeout
distributed.dask_worker - INFO - End worker
Traceback (most recent call last):
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/distributed/cli/dask_worker.py", line 405, in <module>
    go()
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/distributed/cli/dask_worker.py", line 401, in go
    main()
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/distributed/cli/dask_worker.py", line 392, in main
    raise TimeoutError("Timed out starting worker.") from None
tornado.util.TimeoutError: Timed out starting worker.`

I talked to my IT group here and they suggested that I was on the wrong IP address. I did `ifconfig` to checkout which control had the corresponding address and I tried changing the interface to `eth1` (it was using `eth0` before) I ended up getting this:
`Traceback (most recent call last):
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/distributed/cli/dask_worker.py", line 405, in <module>
    go()
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/distributed/cli/dask_worker.py", line 401, in go
    main()
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/distributed/cli/dask_worker.py", line 367, in main
    for i in range(nprocs)
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/distributed/cli/dask_worker.py", line 367, in <listcomp>
    for i in range(nprocs)
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/distributed/nanny.py", line 161, in __init__
    security=security,
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/distributed/comm/addressing.py", line 232, in address_from_user_args
    host = get_ip_interface(interface)
  File "/homes/user/miniconda3/envs/py36/lib/python3.6/site-packages/distributed/utils.py", line 173, in get_ip_interface
    for info in psutil.net_if_addrs()[ifname]:
KeyError: 'eth1'`

Any thoughts on how I should proceed?

**Comment by kyrubas** [comment]
Ok, so I figured it out; the issue turned out to be I need to use the `ip` arg rather than pointing to the interface with the `interface` arg. Thanks you your help!

**Comment by guillaumeeb** [comment]
Glad you found the solution!

Happy Dasking!

**Comment by Cipcu** [comment]
Could you elaborate a bit more? I have the same error, I don't use `cluster = SLURMCluster(...` however, but `cluster = KubeCluster.from_yaml('worker-spec.yml')`, the logs from the pods look quite the same tho. Where did you use the IP arg, if I may ask. Thanks!

**Comment by kyrubas** [comment]
When creating a cluster object you might need to explicitly call out to the IP address of the scheduler. I suspect under a yaml file under something like:

```
jobqueue:
  KubeCluster:
    ip: X.X.X.X
    ...
```
Here's what I've been doing (in a jupyter notebook) when testing out dask-jobqueue
```
cluster = SLURMCluster(cores=1, processes=1, 
                  memory='1GB', 
                  project='testing', 
                  queue="testqueue", 
                  ip="X.X.X.X",
                  log_directory="./dasklogs", 
                  local_directory="./dasklocal")
```

**Comment by Cipcu** [comment]
Thanks for your reply. I fixed it by downgrading Tornado from 5.1.1. to 5.0.0. Now everything seem to work fine.

Have a nice day :)
